File: examples/example-kubernetes-client-openapi-reactor-java/src/main/java/micronaut/client/PodController.java
package micronaut.client;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.kubernetes.client.openapi.reactor.api.CoreV1ApiReactor;
import jakarta.inject.Inject;
import jakarta.validation.constraints.NotNull;
import reactor.core.publisher.Mono;

import java.util.Map;
import java.util.stream.Collectors;

@Controller("/pods")
public class PodController {

    @Inject
    CoreV1ApiReactor coreV1ApiReactor;

    @Get("/{namespace}/{name}")
    public Mono<String> getPod(final @NotNull String namespace, final @NotNull String name) {
        return coreV1ApiReactor.readNamespacedPod(name, namespace, null)
            .map(it -> it.getStatus().getPhase());
    }

    @Get("/{namespace}")
    public Mono<Map<String, String>> getPods(final @NotNull String namespace) {
        return coreV1ApiReactor.listNamespacedPod(namespace, null, null, null, null, null, null, null, null, null, null, null, null)
            .map(it -> it.getItems().stream()
                .filter(p -> p.getStatus() != null)
                .collect(Collectors.toMap(
                    p -> p.getMetadata().getName(),
                    p -> p.getStatus().getPhase())));
    }
}

File: examples/example-kubernetes-client-openapi-reactor-python/src/main/python/micronaut/client/PodController.py
from typing import Annotated

from jakarta.inject import Inject
from jakarta.validation.constraints import NotNull
from micronaut.http.annotation import Controller, Get
from micronaut.kubernetes.client.openapi.reactor.api import CoreV1ApiReactor
from reactor.core.publisher import Mono


@Controller("/pods")
class PodController:

    core_v1_api_reactor: Annotated[CoreV1ApiReactor, Inject]

    @Get("/{namespace}/{name}")
    def get_pod(self, namespace: Annotated[str, NotNull], name: Annotated[str, NotNull]) -> Mono[str]:
        return self.core_v1_api_reactor.readNamespacedPod(name, namespace, None) \
            .map(lambda it: it.getStatus().getPhase())

    @Get("/{namespace}")
    def get_pods(self, namespace: Annotated[str, NotNull]) -> Mono[dict[str, str]]:
        return self.core_v1_api_reactor.listNamespacedPod(namespace, None, None, None, None, None, None, None, None, None, None, None, None) \
            .map(lambda it: {
                p.getMetadata().getName(): p.getStatus().getPhase()
                for p in it.getItems()
                if p.getStatus() is not None
            })
